File: src/java-sample/com/xtaudio/xt/RenderDefaultOutput.java
package com.xtaudio.xt;

import com.sun.jna.Pointer;
import java.util.Random;

public class RenderDefaultOutput {

    static final Random RANDOM = new Random();

    static void render(XtStream stream, Object input, Object output, int frames,
            double time, long position, boolean timeValid, long error, Object user) {

        XtFormat format = stream.getFormat();
        short[] buffer = (short[]) output;
        for (int f = 0; f < frames; f++)
            for (int c = 0; c < format.outputs; c++) {
                double noise = RANDOM.nextDouble() * 2.0 - 1.0;
                buffer[f * format.outputs + c] = (short) (noise * Short.MAX_VALUE);
            }
    }

    public static void main(String[] args) {

        try (XtAudio audio = new XtAudio(null, Pointer.NULL, null, null)) {
            XtService service = XtAudio.getServiceBySetup(XtSetup.CONSUMER_AUDIO);
            try (XtDevice device = service.openDefaultDevice(true)) {

                if (device == null) {
                    System.out.println("No default device found.");
                    return;
                }

                XtFormat format = new XtFormat(new XtMix(44100, XtSample.INT16), 0, 0, 2, 0);
                if (!device.supportsFormat(format)) {
                    System.out.println("Format not supported.");
                    return;
                }

                XtBuffer buffer = device.getBuffer(format);
                try (XtStream stream = device.openStream(format, true, false, buffer.current, RenderDefaultOutput::render, "user-data")) {
                    stream.start();
                    Thread.sleep(1000);
                    stream.stop();
                } catch (InterruptedException e) {
                    throw new RuntimeException(e);
                }
            }
        }
    }
}

File: src/cli-sample/RenderDefaultOutput.cs
using System;
using System.Threading;

namespace Xt {

    public static class RenderDefaultOutput {

        static readonly Random Rand = new Random();

        static void Render(XtStream stream, object input, object output, int frames,
                double time, ulong position, bool timeValid, ulong error, Object user) {

            short[] buffer = (short[])output;
            XtFormat format = stream.GetFormat();
            for (int f = 0; f < frames; f++)
                for (int c = 0; c < format.outputs; c++) {
                    double noise = Rand.NextDouble() * 2.0 - 1.0;
                    buffer[f * format.outputs + c] = (short)(noise * short.MaxValue);
                }
        }

        [STAThread]
        public static void Main(string[] args) {

            using (XtAudio audio = new XtAudio(null, IntPtr.Zero, null, null)) {
                XtService service = XtAudio.GetServiceBySetup(XtSetup.ConsumerAudio);
                using (XtDevice device = service.OpenDefaultDevice(true)) {

                    if (device == null) {
                        Console.WriteLine("No default device found.");
                        return;
                    }

                    XtFormat format = new XtFormat(new XtMix(44100, XtSample.Int16), 0, 0, 2, 0);
                    if (!device.SupportsFormat(format)) {
                        Console.WriteLine("Format not supported.");
                        return;
                    }

                    XtBuffer buffer = device.GetBuffer(format);
                    using (XtStream stream = device.OpenStream(format, true, false, buffer.current, Render, "user-data")) {
                        stream.Start();
                        Thread.Sleep(1000);
                        stream.Stop();
                    }
                }
            }
        }
    }
}
